### PR TITLE
Add rspec for bets_controller

### PR DESCRIPTION
**Pull request by ghost**
Added rspec for controllers/bets_controller.rb

### CODE REVIEW

File: spec/controllers/bets_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    let(:invalid_params) do
+      { bet: 'Invalid Bet',
+        expires_at: DateTime.yesterday,
+        creator_id: user.id }
+    end
+    subject { post :create, params: { bet: bet_params } }
+    context 'with valid params' do
+      let(:bet_params) { valid_params }
+      let(:created_bet) { Bet.last }
+      it { expect { subject }.to change(Bet, :count).by(1) }
```
Comment (swively) [inline spec/controllers/bets_controller_spec.rb]: when you aren't able to write `it { is_expected ... ` in a way that makes sense when reading it back, it is preferrable to write it out on multiplte lines:
```
it 'changes the bet count' do
  expect(subject).to change(etc)
end
```

This is also a situation where naming your subject can helpful
i.e `subject(:get_new) { get :new }` lets you write things like `expect(get_new).to do_something_cool`

**Review by swively — DISMISSED** [review]

File: spec/controllers/bets_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      { bet: 'Bet text',
+        expires_at: DateTime.tomorrow,
+        creator_id: user.id }
+    end
+    let(:invalid_params) do
+      { bet: 'Invalid Bet',
+        expires_at: DateTime.yesterday,
+        creator_id: user.id }
+    end
+    subject { post :create, params: { bet: bet_params } }
```
Comment (qr8r) [inline spec/controllers/bets_controller_spec.rb]: This is a style comment, but I prefer the subject to be the first thing in the test, so its easy to find exactly what is being tested.

```ruby 
describe 'thing' do
  subject { ... }
end
```

File: spec/controllers/bets_controller_spec.rb (hunk 1 above)
Comment (ghost) [reply spec/controllers/bets_controller_spec.rb]: Thanks, Mike, hopefully I fixed it now.

**Comment by cbrunsdon** [comment]
jenkins test this please

File: spec/controllers/bets_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        expires_at: DateTime.tomorrow,
+        creator_id: user.id }
+    end
+    let(:invalid_params) do
+      { bet: 'Invalid Bet',
+        expires_at: DateTime.yesterday,
+        creator_id: user.id }
+    end
+    context 'with valid params' do
+      let(:bet_params) { valid_params }
```
Comment (qr8r) [inline spec/controllers/bets_controller_spec.rb]: Why not define this directly here

**Review by qr8r — CHANGES_REQUESTED** [review]

File: spec/controllers/bets_controller_spec.rb (hunk 3 above)
Comment (ghost) [reply spec/controllers/bets_controller_spec.rb]: Thanks! Should be fixed now.

**Review by qr8r — APPROVED** [review]

File: spec/controllers/bets_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    it { is_expected.to be_ok }
+  end
+
+  describe 'GET #new' do
+    subject { get :new }
+    it { is_expected.to be_ok }
+  end
+
+  describe 'POST #create' do
+    subject(:post_create) { post :create, params: { bet: bet_params } }
```
Comment (jacquesporveau) [inline spec/controllers/bets_controller_spec.rb:28]: In all your other specs above this you aren't overwriting the `subject`'s name but in this case you are. Any reasoning behind that? I would like to see a more consistent method being taken.

Comment (jacquesporveau) [reply spec/controllers/bets_controller_spec.rb:28]: Yarek has told me in person the reason he made these changes and they make sense to me.

**Comment by jacquesporveau** [comment]
This branch is out of date with the base branch. Please update.

**Comment by qr8r** [comment]
Needs rebase

**Pull request merged**